### PR TITLE
Civ 6055

### PR DESCRIPTION
**Pull request by hgulv1**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/CIV-6055

### Change description ###
Setting a more descriptive error message for a nullpointer exception occuring when responding to a claim

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x ] No
```

### CODE REVIEW

**Review by darrenHMCTS — DISMISSED** [review]

**Review by jeejavr — CHANGES_REQUESTED** [review]

**Review by asthamalviya — CHANGES_REQUESTED** [review]
I wonder how primary address missing is a BAD_REQUEST error?

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/RespondToClaimCallbackHandler.java [hunk 1]
```diff
@@ -351,6 +353,10 @@ private CallbackResponse resetStatementOfTruth(CallbackParams callbackParams) {
     private CallbackResponse setApplicantResponseDeadline(CallbackParams callbackParams) {
         CaseData caseData = callbackParams.getCaseData();
 
+        if (caseData.getRespondent1Copy().getPrimaryAddress() == null) {
+            throw new HttpClientErrorException(HttpStatus.BAD_REQUEST, "Primary Address cannot be empty");
+        }
+
```
[the remark this answers is not in the document]
Comment (hgulv1) [reply src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/RespondToClaimCallbackHandler.java:357]: spoke offline and agreed on changes

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/RespondToClaimCallbackHandler.java [hunk 2]
```diff
@@ -351,6 +353,10 @@ private CallbackResponse resetStatementOfTruth(CallbackParams callbackParams) {
     private CallbackResponse setApplicantResponseDeadline(CallbackParams callbackParams) {
         CaseData caseData = callbackParams.getCaseData();
 
+        if (caseData.getRespondent1Copy().getPrimaryAddress() == null) {
+            throw new HttpClientErrorException(HttpStatus.BAD_REQUEST, "Primary Address cannot be empty");
```
[the remark this answers is not in the document]
Comment (hgulv1) [reply src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/RespondToClaimCallbackHandler.java]: new changes will solve this

**Review by asthamalviya — CHANGES_REQUESTED** [review]
In this case the code will simply not run if information is missing. While we need is throw a proper error in case we come across to null pointer exception.

**Review by asthamalviya — APPROVED** [review]

**Review by jeejavr — APPROVED** [review]

**Pull request merged**